File: 1 - Iniciante/C99/BEE1118.c
#include <stdio.h>

int main(){
	float n1, n2, media;
	
	while(1){
		
		//Entrada de dados c/ validação
		do{
			scanf("%f", &n1);
		
			if (n1 < 0 || n1 > 10) printf("nota invalida\n");
		} while (n1 < 0 || n1 > 10);
	
		do{
			scanf("%f", &n2);
		
			if (n2 < 0 || n2 > 10) printf("nota invalida\n");
		}while (n2 < 0 || n2 > 10); 
	
		//Média
		media = (n1 + n2) / 2;
		printf("media = %.2f\n", media);
		
		//Nova operação
		int op;
		
		printf("novo calculo (1-sim 2-nao)\n");
		scanf("%d", &op);
		
		while (1){
			if (op == 1) break;
			else if (op == 2) return 0;
			else{
				printf("novo calculo (1-sim 2-nao)\n");
				scanf("%d", &op);
			}
		}
	}
}

File: 1 - Iniciante/Python 3.11/BEE1118.py
novo_calculo = True

while (novo_calculo):
    notas = []
    validas = 0
    
    while (validas<2):
        nota = float(input())

        if (nota<0 or nota>10):
            print("nota invalida")
            continue
        
        notas.append(nota)
        validas += 1

    print(f'media = {(notas[0] + notas[1])/2:.2f}')
    
    while (True):
        print("novo calculo (1-sim 2-nao)")
        opcao = int(input())
        match (opcao):
            case 1:
                novo_calculo = True
                break
            case 2:
                novo_calculo = False
                break
            case _:
                continue